Library/Util.ts

```diff
@@ -151,5 +154,29 @@ class Util {
 
         return map;
     }
+
+    /**
+     * Checks if a request url is not on a excluded domain list 
+     * and if it is safe to add correlation headers (x-ms-request-source-ikey, x-ms-request-target-ikey)
+     */
+    public static canIncludeCorrelationHeader(client: Client, options: http.RequestOptions | http.ServerResponse) {
+        let excludedDomains = client && client.config && client.config.correlationHeaderExcludedDomains;
+        if (!excludedDomains || excludedDomains.length == 0 || !options) {
+            return true;
+        }
+
+        let url = (<http.RequestOptions>options).host ||
+                ((<http.ServerResponse>options).getHeader && (<http.ServerResponse>options).getHeader("host"));
```

Comment:
I see where it is called from, but I don't see how it's going to work because there is no such `"host"` header for a server response as far as I know. (Or if there is, I guess it would refer to the server host, which doesn't make sense to filter on since it is always the same.)